lib/AuthStateManager.ts

```diff
@@ -131,7 +140,7 @@ class AuthStateManager {
             isPending = shouldEvaluateIsPending();
           }
           let promise = isAuthenticated 
-            ? isAuthenticated(this._sdk)
+            ? isAuthenticated(this._sdk, isPending)
```

Comment:
why is `isPending` being passed to `isAuthenticated`? The method could definitely benefit from having the entire `authState` object which has all properties set except `isAuthenticated`. Then `isAuthenticated` can have simple logic like 

```
return (authState.accessToken && authState.idToken)
```

without needing to do async logic or check expired. This logic has already been done.

Reply:
`isAuthenticated` is also part of the `authState`, seems like what we want is new tokens with old isAuthencated state, which will cause more confusion. I am ok with adding new tokens to the params, or just remove the second input.